// @deere/fuel-react/src/fuel-lab/Accordion/AccordionItem.js
import React, { useContext, forwardRef } from "react";
import { withStyles } from '@material-ui/core/styles';
import { Accordion as MuiAccordion, AccordionSummary as MuiAccordionSummary, AccordionDetails, Typography, Box } from '@material-ui/core';
import ChevronRight from '@material-ui/icons/ChevronRight';
import classNames from 'classnames';
import styles from './Accordion.module.scss';
import { useIds } from '../';
import AccordionContext from './AccordionContext';

const Accordion = withStyles({
  root: {
    '&$expanded': {
      backgroundColor: '#fff',
    },
  },
  expanded: {},
})(MuiAccordion);

const AccordionSummary = withStyles({
  content: {
    margin: '.5rem 0'
  },
  expanded: {
    "& .MuiSvgIcon-root": {
      transform: "rotate(90deg)"
    }
  },
})(MuiAccordionSummary);


export const AccordionItem = forwardRef(({ className, id, open: openProp, defaultOpen = false, children, title, summary, ...props }, ref) => {
  const [buttonId, panelId] = useIds(id, `accordion-button`, `accordion-panel`);
  const contextEventKey = useContext(AccordionContext);

  const [open, setOpen] = React.useState(
    openProp ? true : defaultOpen
  );

  const handleAccChange = (id) => (event, newExpanded) => {
    console.log(newExpanded);
    contextEventKey.handleChange(id, newExpanded);
    console.log(contextEventKey.activeKey);
    setOpen(contextEventKey.activeKey === id);
  };

  return (
    <Accordion ref={ref} expanded={open} onChange={handleAccChange(buttonId)} className={classNames(styles.accordionItem, className)} {...props}>
      <AccordionSummary className={styles.accordionSummary}
        aria-controls={panelId}
        id={buttonId}
      >
        <ChevronRight className={styles.accordionExpandIcon} />
        <Typography className={styles.accordionTitle}>{title}</Typography>
        {summary && <Box ml="auto" mr=".75rem"><Typography className={styles.accordionSummaryTitle}>{summary}</Typography></Box>}
      </AccordionSummary>
      <AccordionDetails className={styles.accordionDetails}>
        <Typography>
          {children}
        </Typography>
      </AccordionDetails>
    </Accordion>
  );
});
