// @deere/fuel-react/src/Checkbox/Checkbox.tsx
import React from "react";
import { withStyles } from '@material-ui/core/styles';
import MaterialCheckbox from '@material-ui/core/Checkbox';
import { CheckboxProps as MaterialCheckboxProps } from '@material-ui/core/Checkbox';

export interface CheckboxProps extends Omit<MaterialCheckboxProps,
  'disableFocusRipple' |
  'centerRipple' |
  'disableTouchRipple' |
  'focusRipple' |
  'disableRipple' |
  'TouchRippleProps' |
  'color'
> {
}

const StyledCheckbox = withStyles((theme) => ({
  root: {
    height: "1.875rem",
    width: "2.125rem",
    marginLeft: theme.spacing(1),
    color: 'var(--fuel-color-border-dark)',
    "&.Mui-focusVisible .MuiSvgIcon-root": {
      borderRadius: theme.shape.borderRadius,
      boxShadow: 'inset 0 0 0 0.125rem var(--fuel-color-focused)'
    },
    '&:hover': {
      backgroundColor: 'transparent'
    },
    '&.Mui-checked:hover': {
      backgroundColor: 'transparent'
    },
    '&.MuiIconButton-edgeStart': {
      marginLeft: '-12px'
    },
    '&.MuiIconButton-edgeEnd': {
      marginRight: '-12px'
    }
  },
  indeterminate: {
    color: 'var(--fuel-color-background-interactive-default-base)'
  },

}))(MaterialCheckbox);

export const Checkbox = React.forwardRef<any, CheckboxProps>(
  (
    {
      ...props
    },
    ref
  ) => {
    return (
      <StyledCheckbox ref={ref} {...props} />
    );
  });
