// @deere/fuel-react/src/Select/NativeSelect.tsx
import React from "react";
import MaterialNativeSelect from '@material-ui/core/NativeSelect';
import { withStyles } from '@material-ui/core/styles';
import { Input } from '../';
import { NativeSelectProps as MaterialNativeSelectProps } from '@material-ui/core/NativeSelect';

export interface NativeSelectProps extends MaterialNativeSelectProps { };

const StyledNativeSelect = withStyles((theme) => ({
  select: {
    borderRadius: theme.shape.borderRadius,
    '&:focus': {
      borderRadius: theme.shape.borderRadius,
    },
    paddingRight: '1.75rem !important',
    lineHeight: '1.5rem'
  },
  icon: {
    width: '.875em',
    height: '.875em',
    right: '.375rem',
    top: 'calc(50% - 11px)',
    color: 'currentColor'
  },
  iconOpen: {
    transform: 'none'
  }
}))(MaterialNativeSelect);

export const NativeSelect = React.forwardRef<any, NativeSelectProps>(
  (
    {
      children,
      ...props
    },
    ref
  ) => {
    return (
      <StyledNativeSelect input={<Input />} ref={ref} {...props}>
        {children}
      </StyledNativeSelect>
    );
  });
