// @deere/fuel-react/src/Select/Select.tsx
import React from "react";
import MaterialSelect from '@material-ui/core/Select';
import { withStyles } from '@material-ui/core/styles';
import { Input } from '../';
import KeyboardArrowDownIcon from '@material-ui/icons/KeyboardArrowDown';
import { SelectProps as MaterialSelectProps } from '@material-ui/core/Select';
import { MenuProps } from '../Menu';

export interface SelectProps extends MaterialSelectProps { };

const StyledSelect = withStyles((theme) => ({
  select: {
    borderRadius: theme.shape.borderRadius,
    '&:focus': {
      borderRadius: theme.shape.borderRadius,
    },
    paddingRight: '1.75rem !important',
    height: '1.625rem',
    display: 'inline-flex',
    alignItems: 'center',
    lineHeight: '1.5rem',
    "&[disabled]": {
      cursor: "not-allowed",
      backgroundColor: 'var(--fuel-color-gray-100)',
      borderColor: 'var(--fuel-color-border-light)'
    }
  },
  icon: {
    width: '.875em',
    height: '.875em',
    right: '.375rem',
    top: 'calc(50% - 11px)',
    color: 'currentColor'
  },
  iconOpen: {
    transform: 'none'
  }
}))(MaterialSelect);

const defaultMenuProps = {
  anchorOrigin: {
    vertical: "bottom",
    horizontal: "left",
  },
  transformOrigin: { vertical: 'top', horizontal: 'left' },
  getContentAnchorEl: null,
  disableScrollLock: true
}

export const Select = React.forwardRef<any, SelectProps>(
  (
    {
      children,
      MenuProps = defaultMenuProps as MenuProps,
      ...props
    },
    ref
  ) => {
    return (
      <StyledSelect
        input={<Input />}
        IconComponent={KeyboardArrowDownIcon}
        ref={ref}
        MenuProps={MenuProps}
        {...props}
      >
        {children}
      </StyledSelect>
    );
  });
