// @deere/fuel-react/lib/themes/baseOverrides.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.baseOverrides = void 0;

(function () {
  var enterModule = typeof reactHotLoaderGlobal !== 'undefined' ? reactHotLoaderGlobal.enterModule : undefined;
  enterModule && enterModule(module);
})();

var __signature__ = typeof reactHotLoaderGlobal !== 'undefined' ? reactHotLoaderGlobal["default"].signature : function (a) {
  return a;
};

var baseOverrides = {
  MuiCssBaseline: {
    '@global': {
      body: {
        fontSize: 'var(--fuel-font-size-base)'
      }
    }
  },
  MuiScopedCssBaseline: {
    root: {
      fontSize: 'var(--fuel-font-size-base)'
    }
  },
  MuiSwitch: {
    root: {
      width: 42,
      height: 26,
      padding: 0,
      margin: 8
    },
    switchBase: {
      padding: 1,
      '&$checked, &$colorPrimary$checked, &$colorSecondary$checked': {
        transform: 'translateX(16px)',
        color: '#fff',
        '& + $track': {
          opacity: 1,
          border: 'none'
        }
      }
    },
    thumb: {
      width: 24,
      height: 24
    },
    track: {
      borderRadius: 13,
      border: '1px solid #bdbdbd',
      backgroundColor: '#fafafa',
      opacity: 1,
      transition: 'background-color 300ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,border 300ms cubic-bezier(0.4, 0, 0.2, 1) 0ms'
    }
  }
};
exports.baseOverrides = baseOverrides;
;

(function () {
  var reactHotLoader = typeof reactHotLoaderGlobal !== 'undefined' ? reactHotLoaderGlobal.default : undefined;

  if (!reactHotLoader) {
    return;
  }

  reactHotLoader.register(baseOverrides, "baseOverrides", "/var/jenkins_home/workspace/ish_Fuel_React_Components_master/src/themes/baseOverrides.ts");
})();

;

(function () {
  var leaveModule = typeof reactHotLoaderGlobal !== 'undefined' ? reactHotLoaderGlobal.leaveModule : undefined;
  leaveModule && leaveModule(module);
})();
