// @deere/fuel-react/lib/themes/baseTheme.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.baseTheme = void 0;

var _styles = require("@material-ui/core/styles");

var _tokens = require("@deere/fuel-tokens/dist/web/jd-green/tokens");

(function () {
  var enterModule = typeof reactHotLoaderGlobal !== 'undefined' ? reactHotLoaderGlobal.enterModule : undefined;
  enterModule && enterModule(module);
})();

var __signature__ = typeof reactHotLoaderGlobal !== 'undefined' ? reactHotLoaderGlobal["default"].signature : function (a) {
  return a;
};

var baseTheme = (0, _styles.createMuiTheme)({
  typography: {
    fontFamily: _tokens.FuelFontFamilyBase,
    fontWeightMedium: _tokens.FuelFontWeightSemibold,
    h1: {
      fontWeight: _tokens.FuelFontWeightSemibold,
      fontSize: _tokens.FuelFontSizeHeading1,
      lineHeight: _tokens.FuelLineHeight2xl
    },
    h2: {
      fontWeight: _tokens.FuelFontWeightSemibold,
      fontSize: _tokens.FuelFontSizeHeading2,
      lineHeight: _tokens.FuelLineHeightXl
    },
    h3: {
      fontWeight: _tokens.FuelFontWeightSemibold,
      fontSize: _tokens.FuelFontSizeHeading3,
      lineHeight: _tokens.FuelLineHeightXl
    },
    h4: {
      fontWeight: _tokens.FuelFontWeightSemibold,
      fontSize: _tokens.FuelFontSizeHeading4,
      lineHeight: _tokens.FuelLineHeightMd
    },
    h5: {
      fontWeight: _tokens.FuelFontWeightSemibold,
      fontSize: _tokens.FuelFontSizeHeading5,
      lineHeight: _tokens.FuelLineHeightSm
    },
    h6: {
      fontWeight: _tokens.FuelFontWeightSemibold,
      fontSize: _tokens.FuelFontSizeHeading6,
      lineHeight: _tokens.FuelLineHeightSm
    },
    body1: {
      fontSize: _tokens.FuelFontSizeMd
    },
    subtitle1: {
      color: _tokens.FuelColorTextTertiary,
      fontWeight: _tokens.FuelFontWeightSemibold,
      fontSize: _tokens.FuelFontSizeSm,
      lineHeight: _tokens.FuelLineHeightSm
    },
    subtitle2: {
      fontWeight: _tokens.FuelFontWeightSemibold,
      color: _tokens.FuelColorTextTertiary,
      fontSize: _tokens.FuelFontSizeXs,
      lineHeight: _tokens.FuelLineHeightXs
    },
    button: {
      textTransform: 'none',
      fontSize: _tokens.FuelFontSizeSm,
      lineHeight: _tokens.FuelLineHeightMd,
      fontWeight: _tokens.FuelFontWeightSemibold
    },
    caption: {
      fontSize: _tokens.FuelFontSizeXs,
      lineHeight: _tokens.FuelLineHeightXs,
      fontWeight: _tokens.FuelFontWeightRegular
    },
    overline: {
      fontSize: _tokens.FuelFontSizeXs,
      lineHeight: _tokens.FuelLineHeightXs,
      fontWeight: _tokens.FuelFontWeightRegular
    }
  },
  props: {
    MuiButtonBase: {
      disableRipple: true
    },
    MuiButtonGroup: {
      disableRipple: true
    },
    MuiLink: {
      underline: "always"
    }
  },
  shape: {
    borderRadius: 4
  },
  spacing: function spacing(factor) {
    return "".concat(0.25 * factor, "rem");
  },
  shadows: ["none", _tokens.FuelShadowMd, _tokens.FuelShadowMd, _tokens.FuelShadowMd, _tokens.FuelShadowMd, _tokens.FuelShadowLg, _tokens.FuelShadowLg, _tokens.FuelShadowLg, _tokens.FuelShadowXl, _tokens.FuelShadowXl, _tokens.FuelShadowXl, _tokens.FuelShadowXl, _tokens.FuelShadowXl, _tokens.FuelShadowXl, _tokens.FuelShadowXl, _tokens.FuelShadowXl, _tokens.FuelShadow2xl, _tokens.FuelShadow2xl, _tokens.FuelShadow2xl, _tokens.FuelShadow2xl, _tokens.FuelShadow2xl, _tokens.FuelShadow2xl, _tokens.FuelShadow2xl, _tokens.FuelShadow2xl, _tokens.FuelShadow2xl]
});
exports.baseTheme = baseTheme;
;

(function () {
  var reactHotLoader = typeof reactHotLoaderGlobal !== 'undefined' ? reactHotLoaderGlobal.default : undefined;

  if (!reactHotLoader) {
    return;
  }

  reactHotLoader.register(baseTheme, "baseTheme", "/var/jenkins_home/workspace/ish_Fuel_React_Components_master/src/themes/baseTheme.ts");
})();

;

(function () {
  var leaveModule = typeof reactHotLoaderGlobal !== 'undefined' ? reactHotLoaderGlobal.leaveModule : undefined;
  leaveModule && leaveModule(module);
})();
