// @deere/fuel-react/lib/themes/jd-black.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.jdBlack = exports.jdBlackOverrides = void 0;

var _styles = require("@material-ui/core/styles");

var _baseTheme = require("./baseTheme");

var _baseOverrides = require("./baseOverrides");

var _tokens = require("@deere/fuel-tokens/dist/web/jd-black/tokens");

require("!style-loader!css-loader!@deere/fuel-tokens/dist/web/jd-black/fuel-theme-jd-black.css");

(function () {
  var enterModule = typeof reactHotLoaderGlobal !== 'undefined' ? reactHotLoaderGlobal.enterModule : undefined;
  enterModule && enterModule(module);
})();

function ownKeys(object, enumerableOnly) { var keys = Object.keys(object); if (Object.getOwnPropertySymbols) { var symbols = Object.getOwnPropertySymbols(object); if (enumerableOnly) { symbols = symbols.filter(function (sym) { return Object.getOwnPropertyDescriptor(object, sym).enumerable; }); } keys.push.apply(keys, symbols); } return keys; }

function _objectSpread(target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i] != null ? arguments[i] : {}; if (i % 2) { ownKeys(Object(source), true).forEach(function (key) { _defineProperty(target, key, source[key]); }); } else if (Object.getOwnPropertyDescriptors) { Object.defineProperties(target, Object.getOwnPropertyDescriptors(source)); } else { ownKeys(Object(source)).forEach(function (key) { Object.defineProperty(target, key, Object.getOwnPropertyDescriptor(source, key)); }); } } return target; }

function _defineProperty(obj, key, value) { if (key in obj) { Object.defineProperty(obj, key, { value: value, enumerable: true, configurable: true, writable: true }); } else { obj[key] = value; } return obj; }

var __signature__ = typeof reactHotLoaderGlobal !== 'undefined' ? reactHotLoaderGlobal["default"].signature : function (a) {
  return a;
};

var jdBlackOverrides = {
  MuiBadge: {
    colorPrimary: {
      backgroundColor: _tokens.FuelColorJdYellow,
      color: _tokens.FuelColorBlack
    }
  }
};
exports.jdBlackOverrides = jdBlackOverrides;
var jdBlack = (0, _styles.createMuiTheme)(_objectSpread(_objectSpread({}, _baseTheme.baseTheme), {}, {
  palette: {
    text: {
      primary: _tokens.FuelColorTextPrimary,
      secondary: _tokens.FuelColorTextSecondary,
      disabled: _tokens.FuelColorTextDisabled
    },
    type: 'light',
    primary: {
      main: _tokens.FuelColorBrandPrimary
    },
    secondary: {
      main: _tokens.FuelColorBrandPrimary
    },
    background: {
      "default": _tokens.FuelColorBackgroundGray
    },
    divider: _tokens.FuelColorBorderLight,
    action: {
      hover: _tokens.FuelColorBackgroundActionDefaultHovered,
      selected: _tokens.FuelColorBackgroundActionDefaultSelected
    },
    error: {
      main: '#c21020'
    },
    grey: {
      50: _tokens.FuelColorGray50,
      100: _tokens.FuelColorGray100,
      200: _tokens.FuelColorGray200,
      300: _tokens.FuelColorGray300,
      400: _tokens.FuelColorGray400,
      500: _tokens.FuelColorGray500,
      600: _tokens.FuelColorGray600,
      700: _tokens.FuelColorGray700,
      800: _tokens.FuelColorGray800,
      900: _tokens.FuelColorGray900
    }
  },
  overrides: _objectSpread(_objectSpread({}, _baseOverrides.baseOverrides), jdBlackOverrides)
}));
exports.jdBlack = jdBlack;
;

(function () {
  var reactHotLoader = typeof reactHotLoaderGlobal !== 'undefined' ? reactHotLoaderGlobal.default : undefined;

  if (!reactHotLoader) {
    return;
  }

  reactHotLoader.register(jdBlackOverrides, "jdBlackOverrides", "/var/jenkins_home/workspace/ish_Fuel_React_Components_master/src/themes/jd-black.ts");
  reactHotLoader.register(jdBlack, "jdBlack", "/var/jenkins_home/workspace/ish_Fuel_React_Components_master/src/themes/jd-black.ts");
})();

;

(function () {
  var leaveModule = typeof reactHotLoaderGlobal !== 'undefined' ? reactHotLoaderGlobal.leaveModule : undefined;
  leaveModule && leaveModule(module);
})();
