// @deere/ux.uxframe-react/lib/components/Search/SearchClearButton.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.SearchClearButton = void 0;

var _react = _interopRequireDefault(require("react"));

var _classnames = _interopRequireDefault(require("classnames"));

var _propTypes = _interopRequireDefault(require("prop-types"));

var _Button = require("../Button");

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _defineProperty(obj, key, value) { if (key in obj) { Object.defineProperty(obj, key, { value: value, enumerable: true, configurable: true, writable: true }); } else { obj[key] = value; } return obj; }

function _extends() { _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; }; return _extends.apply(this, arguments); }

function _objectWithoutProperties(source, excluded) { if (source == null) return {}; var target = _objectWithoutPropertiesLoose(source, excluded); var key, i; if (Object.getOwnPropertySymbols) { var sourceSymbolKeys = Object.getOwnPropertySymbols(source); for (i = 0; i < sourceSymbolKeys.length; i++) { key = sourceSymbolKeys[i]; if (excluded.indexOf(key) >= 0) continue; if (!Object.prototype.propertyIsEnumerable.call(source, key)) continue; target[key] = source[key]; } } return target; }

function _objectWithoutPropertiesLoose(source, excluded) { if (source == null) return {}; var target = {}; var sourceKeys = Object.keys(source); var key, i; for (i = 0; i < sourceKeys.length; i++) { key = sourceKeys[i]; if (excluded.indexOf(key) >= 0) continue; target[key] = source[key]; } return target; }

var ClearIcon = function ClearIcon(props) {
  return /*#__PURE__*/_react["default"].createElement("svg", props, /*#__PURE__*/_react["default"].createElement("path", {
    d: "M19 6.41L17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12z"
  }), /*#__PURE__*/_react["default"].createElement("path", {
    d: "M0 0h24v24H0z",
    fill: "none"
  }));
};

ClearIcon.defaultProps = _defineProperty({
  focusable: "false",
  'aria-hidden': "true",
  fill: "#666",
  xmlns: "http://www.w3.org/2000/svg",
  width: "24",
  height: "24",
  viewBox: "0 0 24 24"
}, "focusable", 'false');

var SearchClearButton = function SearchClearButton(_ref) {
  var ariaLabel = _ref.ariaLabel,
      className = _ref.className,
      props = _objectWithoutProperties(_ref, ["ariaLabel", "className"]);

  var classes = (0, _classnames["default"])('uxf-search-bar-clear-icon', className);
  return /*#__PURE__*/_react["default"].createElement(_Button.Button, _extends({
    variant: "uxf-icon",
    type: "reset",
    "aria-label": ariaLabel,
    className: classes
  }, props), /*#__PURE__*/_react["default"].createElement(ClearIcon, {
    className: "uxf-icon-remove"
  }));
};

exports.SearchClearButton = SearchClearButton;
SearchClearButton.propTypes = {
  /**
   *  Aria-label of clear button. This notifies individuals using a screen
   *  reader that they are on the clear search component.
   */
  ariaLabel: _propTypes["default"].string.isRequired
};
