// @deere/fuel-react/lib/fuel-lab/Accordion/Accordion.stories.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.Custom2 = exports.Custom = exports.Default = exports["default"] = void 0;

var _react = _interopRequireDefault(require("react"));

var _ = require("./");

var _core = require("@material-ui/core");

var _Typography = require("../../Typography/Typography");

var _ExpandMore = _interopRequireDefault(require("@material-ui/icons/ExpandMore"));

var _jsxRuntime = require("@emotion/react/jsx-runtime");

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

(function () {
  var enterModule = typeof reactHotLoaderGlobal !== 'undefined' ? reactHotLoaderGlobal.enterModule : undefined;
  enterModule && enterModule(module);
})();

var __signature__ = typeof reactHotLoaderGlobal !== 'undefined' ? reactHotLoaderGlobal["default"].signature : function (a) {
  return a;
};

var _default = {
  title: 'Components/Accordion',
  component: _.Accordion
};
var _default2 = _default;
exports["default"] = _default2;

var Default = function Default() {
  return (0, _jsxRuntime.jsxs)(_jsxRuntime.Fragment, {
    children: [(0, _jsxRuntime.jsxs)(_core.Accordion, {
      children: [(0, _jsxRuntime.jsx)(_.AccordionSummary, {
        expandIcon: (0, _jsxRuntime.jsx)(_ExpandMore["default"], {}),
        "aria-controls": "panel1a-content",
        id: "panel1a-header",
        children: (0, _jsxRuntime.jsx)(_Typography.Typography, {
          children: "Accordion 1"
        })
      }), (0, _jsxRuntime.jsx)(_.AccordionDetails, {
        children: (0, _jsxRuntime.jsx)(_Typography.Typography, {
          children: "Lorem ipsum dolor sit amet, consectetur adipiscing elit. Suspendisse malesuada lacus ex, sit amet blandit leo lobortis eget."
        })
      })]
    }), (0, _jsxRuntime.jsxs)(_core.Accordion, {
      children: [(0, _jsxRuntime.jsx)(_.AccordionSummary, {
        expandIcon: (0, _jsxRuntime.jsx)(_ExpandMore["default"], {}),
        "aria-controls": "panel2a-content",
        id: "panel2a-header",
        children: (0, _jsxRuntime.jsx)(_Typography.Typography, {
          children: "Accordion 2"
        })
      }), (0, _jsxRuntime.jsx)(_.AccordionDetails, {
        children: (0, _jsxRuntime.jsx)(_Typography.Typography, {
          children: "Lorem ipsum dolor sit amet, consectetur adipiscing elit. Suspendisse malesuada lacus ex, sit amet blandit leo lobortis eget."
        })
      })]
    }), (0, _jsxRuntime.jsxs)(_core.Accordion, {
      children: [(0, _jsxRuntime.jsx)(_.AccordionSummary, {
        expandIcon: (0, _jsxRuntime.jsx)(_ExpandMore["default"], {}),
        "aria-controls": "panel3a-content",
        id: "panel3a-header",
        children: (0, _jsxRuntime.jsx)(_Typography.Typography, {
          children: "Accordion 3"
        })
      }), (0, _jsxRuntime.jsx)(_.AccordionDetails, {
        children: (0, _jsxRuntime.jsx)(_Typography.Typography, {
          children: "Lorem ipsum dolor sit amet, consectetur adipiscing elit. Suspendisse malesuada lacus ex, sit amet blandit leo lobortis eget."
        })
      })]
    })]
  });
};

exports.Default = Default;

var Custom = function Custom() {
  return (0, _jsxRuntime.jsxs)(_.Accordion, {
    children: [(0, _jsxRuntime.jsx)(_.AccordionItem, {
      title: "Accordion",
      children: "Lorem ipsum dolor sit amet, consectetur adipiscing elit. Suspendisse malesuada lacus ex, sit amet blandit leo lobortis eget."
    }), (0, _jsxRuntime.jsx)(_.AccordionItem, {
      title: "Accordion 2",
      children: "Lorem ipsum dolor sit amet, consectetur adipiscing elit. Suspendisse malesuada lacus ex, sit amet blandit leo lobortis eget."
    }), (0, _jsxRuntime.jsx)(_.AccordionItem, {
      title: "Accordion 3",
      summary: "Summary",
      children: "Lorem ipsum dolor sit amet, consectetur adipiscing elit. Suspendisse malesuada lacus ex, sit amet blandit leo lobortis eget."
    })]
  });
};

exports.Custom = Custom;

var Custom2 = function Custom2() {
  return (0, _jsxRuntime.jsxs)(_.Accordion, {
    variant: "gray-bg",
    children: [(0, _jsxRuntime.jsx)(_.AccordionItem, {
      defaultOpen: true,
      title: "Accordion",
      children: "Lorem ipsum dolor sit amet, consectetur adipiscing elit. Suspendisse malesuada lacus ex, sit amet blandit leo lobortis eget."
    }), (0, _jsxRuntime.jsx)(_.AccordionItem, {
      title: "Accordion 2",
      children: "Lorem ipsum dolor sit amet, consectetur adipiscing elit. Suspendisse malesuada lacus ex, sit amet blandit leo lobortis eget."
    }), (0, _jsxRuntime.jsx)(_.AccordionItem, {
      title: "Accordion 3",
      summary: "Summary",
      children: "Lorem ipsum dolor sit amet, consectetur adipiscing elit. Suspendisse malesuada lacus ex, sit amet blandit leo lobortis eget."
    })]
  });
};

exports.Custom2 = Custom2;
;

(function () {
  var reactHotLoader = typeof reactHotLoaderGlobal !== 'undefined' ? reactHotLoaderGlobal.default : undefined;

  if (!reactHotLoader) {
    return;
  }

  reactHotLoader.register(Default, "Default", "/var/jenkins_home/workspace/ish_Fuel_React_Components_master/src/fuel-lab/Accordion/Accordion.stories.js");
  reactHotLoader.register(Custom, "Custom", "/var/jenkins_home/workspace/ish_Fuel_React_Components_master/src/fuel-lab/Accordion/Accordion.stories.js");
  reactHotLoader.register(Custom2, "Custom2", "/var/jenkins_home/workspace/ish_Fuel_React_Components_master/src/fuel-lab/Accordion/Accordion.stories.js");
  reactHotLoader.register(_default, "default", "/var/jenkins_home/workspace/ish_Fuel_React_Components_master/src/fuel-lab/Accordion/Accordion.stories.js");
})();

;

(function () {
  var leaveModule = typeof reactHotLoaderGlobal !== 'undefined' ? reactHotLoaderGlobal.leaveModule : undefined;
  leaveModule && leaveModule(module);
})();
