// @deere/fuel-react/src/ThemeProvider/ThemeProvider.tsx
import React from "react";
import { ThemeProvider as MuiThemeProvider } from "@material-ui/core/styles";
import { CssBaseline } from '../';
import { ThemeProviderProps as MaterialThemeProviderProps } from '@material-ui/core/styles';
import { jdGreen, jdBlack } from "../themes";
import { StylesProvider } from '@material-ui/core/styles';
import { Global, css } from '@emotion/react';

export interface ThemeProviderProps extends Omit<MaterialThemeProviderProps,
  'theme'
> {
  /*
    Include CSS Baseline which brings in global CSS to application.
  */
  baseline: boolean,
  /*
    Custom theme attribute used to target Fuel Tokens
  */
  customThemeAttribute?: 'fuel-jd-green' | 'fuel-jd-black' | string,
  /*
    Include font files
  */
  includeFonts: boolean,
  /*
    Theme
  */
  theme: 'jdGreen' | 'jdBlack' | object
}

const getThemeAttribute = (theme: string | object, customThemeAttribute?: string) => {
  switch (theme) {
    case 'jdGreen':
      return 'fuel-jd-green';
    case 'jdBlack':
      return 'fuel-jd-black';
    default:
      return customThemeAttribute;
  }
}

const getTheme = (theme: string | object) => {
  switch (theme) {
    case 'jdGreen':
      return jdGreen;
    case 'jdBlack':
      return jdBlack;
    default:
      return theme;
  }
}

const fontFace = css`
/* Open Sans - Regular v18 vietnamese_latin-ext_latin_greek-ext_greek_cyrillic-ext_cyrillic */
@font-face {
  font-family: 'Open Sans';
  font-style: normal;
  font-weight: 400;
  src: url('https://cdn.ux.deere.com/brand-foundations/1.3.1/typography/open-sans-v18-regular.woff2') format('woff2'),
        url('https://cdn.ux.deere.com/brand-foundations/1.3.1/typography/open-sans-v18-regular.woff') format('woff')
}

/* Open Sans - Semibold v18 vietnamese_latin-ext_latin_greek-ext_greek_cyrillic-ext_cyrillic */
@font-face {
  font-family: 'Open Sans';
  font-style: normal;
  font-weight: 600;
  src: url('https://cdn.ux.deere.com/brand-foundations/1.3.1/typography/open-sans-v18-600.woff2') format('woff2'),
        url('https://cdn.ux.deere.com/brand-foundations/1.3.1/typography/open-sans-v18-600.woff') format('woff')
}

/* Open Sans - Bold v18 vietnamese_latin-ext_latin_greek-ext_greek_cyrillic-ext_cyrillic */
@font-face {
  font-family: 'Open Sans';
  font-style: normal;
  font-weight: 700;
  src: url('https://cdn.ux.deere.com/brand-foundations/1.3.1/typography/open-sans-v18-700.woff2') format('woff2'),
        url('https://cdn.ux.deere.com/brand-foundations/1.3.1/typography/open-sans-v18-700.woff') format('woff')
}
`;


export const ThemeProvider = (
  {
    children,
    baseline = true,
    includeFonts = true,
    theme = 'jdGreen',
    customThemeAttribute,
    ...props
  }: ThemeProviderProps) => {

  const themeAttribute = getThemeAttribute(theme, customThemeAttribute);
  const activeTheme = getTheme(theme);

  return (
    <MuiThemeProvider theme={activeTheme} {...props}>
      {baseline && <CssBaseline />}
      {includeFonts && <Global styles={fontFace} />}
      <StylesProvider injectFirst>
        <div data-theme={themeAttribute}>
          {children}
        </div>
      </StylesProvider>
    </MuiThemeProvider>
  );
};
