// @deere/ux.uxframe-core/src/js/uxframe.with-lib.js
const ComponentsPlus = require('./components/components-plus');

ComponentsPlus.Charts($UXF);

// Had to add '{ parser: { amd: false } }' to webpack config to get this to work properly.
require('datatables.net-bs4/js/dataTables.bootstrap4.js')(window, $);
require('datatables.net-responsive-bs4/js/responsive.bootstrap4.min.js')(window, $);

// Import Chart.js
import 'chart.js/dist/Chart.bundle.min.js';

// Import Flatpikr
import flatpickr from "flatpickr";

// Init Date Pickers
flatpickr(".uxf-date-picker .form-control", {
    dateFormat: "d-M-Y",
    nextArrow: '<svg focusable="false" aria-hidden="true" class="uxf-btn-icon" fill="#666666" xmlns="http://www.w3.org/2000/svg" width="24" height="24" viewBox="0 0 24 24"><path d="M8.59 16.59L13.17 12 8.59 7.41 10 6l6 6-6 6-1.41-1.41z"/><path fill="none" d="M0 0h24v24H0V0z"/></svg>',
    prevArrow: '<svg focusable="false" aria-hidden="true" class="uxf-btn-icon" fill="#666666" xmlns="http://www.w3.org/2000/svg" width="24" height="24" viewBox="0 0 24 24"><path d="M15.41 16.59L10.83 12l4.58-4.59L14 6l-6 6 6 6 1.41-1.41z"/><path fill="none" d="M0 0h24v24H0V0z"/></svg>',
    disableMobile: true
});
flatpickr(".uxf-date-picker.uxf-date-picker-allow-input .form-control", {
    allowInput: true,
    dateFormat: "d-M-Y",
    nextArrow: '<svg focusable="false" aria-hidden="true" class="uxf-btn-icon" fill="#666666" xmlns="http://www.w3.org/2000/svg" width="24" height="24" viewBox="0 0 24 24"><path d="M8.59 16.59L13.17 12 8.59 7.41 10 6l6 6-6 6-1.41-1.41z"/><path fill="none" d="M0 0h24v24H0V0z"/></svg>',
    prevArrow: '<svg focusable="false" aria-hidden="true" class="uxf-btn-icon" fill="#666666" xmlns="http://www.w3.org/2000/svg" width="24" height="24" viewBox="0 0 24 24"><path d="M15.41 16.59L10.83 12l4.58-4.59L14 6l-6 6 6 6 1.41-1.41z"/><path fill="none" d="M0 0h24v24H0V0z"/></svg>',
    disableMobile: true
});
flatpickr(".uxf-date-picker.uxf-date-picker-time .form-control", {
    dateFormat: "d-M-Y H:i",
    enableTime: true,
    nextArrow: '<svg focusable="false" aria-hidden="true" class="uxf-btn-icon" fill="#666666" xmlns="http://www.w3.org/2000/svg" width="24" height="24" viewBox="0 0 24 24"><path d="M8.59 16.59L13.17 12 8.59 7.41 10 6l6 6-6 6-1.41-1.41z"/><path fill="none" d="M0 0h24v24H0V0z"/></svg>',
    prevArrow: '<svg focusable="false" aria-hidden="true" class="uxf-btn-icon" fill="#666666" xmlns="http://www.w3.org/2000/svg" width="24" height="24" viewBox="0 0 24 24"><path d="M15.41 16.59L10.83 12l4.58-4.59L14 6l-6 6 6 6 1.41-1.41z"/><path fill="none" d="M0 0h24v24H0V0z"/></svg>',
    disableMobile: true
});
flatpickr(".uxf-date-picker.uxf-date-picker-range .form-control", {
    dateFormat: "d-M-Y",
    mode: "range",
    nextArrow: '<svg focusable="false" aria-hidden="true" class="uxf-btn-icon" fill="#666666" xmlns="http://www.w3.org/2000/svg" width="24" height="24" viewBox="0 0 24 24"><path d="M8.59 16.59L13.17 12 8.59 7.41 10 6l6 6-6 6-1.41-1.41z"/><path fill="none" d="M0 0h24v24H0V0z"/></svg>',
    prevArrow: '<svg focusable="false" aria-hidden="true" class="uxf-btn-icon" fill="#666666" xmlns="http://www.w3.org/2000/svg" width="24" height="24" viewBox="0 0 24 24"><path d="M15.41 16.59L10.83 12l4.58-4.59L14 6l-6 6 6 6 1.41-1.41z"/><path fill="none" d="M0 0h24v24H0V0z"/></svg>',
    disableMobile: true
});

// Log that loading is finished
$UXF.log('UXFrame Plus JS Loaded');

module.exports = $UXF;
