// @deere/fuel-react/src/Form/FormControlLabel.tsx
import React from "react";
import MaterialFormControlLabel from '@material-ui/core/FormControlLabel';
import { withStyles } from '@material-ui/core/styles';
import { FormControlLabelProps as MaterialFormControlLabelProps } from '@material-ui/core/FormControlLabel';

export interface FormControlLabelProps extends MaterialFormControlLabelProps { };

const StyledFormControlLabel = withStyles((theme) => ({
  root: {
    "&:not(.Mui-disabled):hover": {
      "& .MuiCheckbox-root": {
        "& .MuiSvgIcon-root": {
          fill: theme.palette.primary.main
        }
      },
      "& .MuiRadio-root": {
        "& .MuiSvgIcon-root": {
          fill: theme.palette.primary.main
        }
      }
    },
    "& .MuiCheckbox-root": {
      "& + .MuiFormControlLabel-label": {
        fontSize: theme.typography.button.fontSize,
        lineHeight: '1.125rem'
      }
    },
    "& .MuiRadio-root": {
      "& + .MuiFormControlLabel-label": {
        fontSize: theme.typography.button.fontSize,
        lineHeight: '1.125rem'
      }
    },
    marginBottom: '0'
  }
}))(MaterialFormControlLabel);

export const FormControlLabel = React.forwardRef<any, FormControlLabelProps>(
  (
    {
      children,
      ...props
    },
    ref
  ) => {
    return (
      <StyledFormControlLabel ref={ref} {...props}>
        {children}
      </StyledFormControlLabel>
    );
  });
