// @deere/fuel-react/lib/fuel-lab/Card/Card.stories.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.Default = exports["default"] = void 0;

var _react = _interopRequireDefault(require("react"));

var _ = require(".");

var _core = require("@material-ui/core");

var _jsxRuntime = require("@emotion/react/jsx-runtime");

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

(function () {
  var enterModule = typeof reactHotLoaderGlobal !== 'undefined' ? reactHotLoaderGlobal.enterModule : undefined;
  enterModule && enterModule(module);
})();

var __signature__ = typeof reactHotLoaderGlobal !== 'undefined' ? reactHotLoaderGlobal["default"].signature : function (a) {
  return a;
};

var _default = {
  title: 'Components/Card',
  component: _.Card
};
var _default2 = _default;
exports["default"] = _default2;

var Default = function Default() {
  return (0, _jsxRuntime.jsxs)(_jsxRuntime.Fragment, {
    children: [(0, _jsxRuntime.jsx)(_core.Box, {
      maxWidth: 325,
      children: (0, _jsxRuntime.jsx)(_.Card, {
        children: (0, _jsxRuntime.jsx)(_.CardContent, {
          children: "This is a card."
        })
      })
    }), (0, _jsxRuntime.jsx)(_core.Box, {
      maxWidth: 325,
      mt: 4,
      children: (0, _jsxRuntime.jsxs)(_.Card, {
        children: [(0, _jsxRuntime.jsxs)(_.CardActionArea, {
          children: [(0, _jsxRuntime.jsx)(_.CardMedia, {
            component: "img",
            alt: "Deere equipment",
            height: "140",
            image: "https://www.deere.com/assets/images/common/industries/construction/JD_CE_HE_Group_2_1366x768.jpg",
            title: "Deere equipment"
          }), (0, _jsxRuntime.jsxs)(_.CardContent, {
            children: [(0, _jsxRuntime.jsx)(_core.Typography, {
              gutterBottom: true,
              variant: "h5",
              component: "h2",
              children: "Card Title"
            }), (0, _jsxRuntime.jsx)(_core.Typography, {
              variant: "body2",
              color: "textSecondary",
              component: "p",
              children: "Some quick example text to build on the card title and make up the bulk of the card's content."
            })]
          })]
        }), (0, _jsxRuntime.jsxs)(_.CardActions, {
          children: [(0, _jsxRuntime.jsx)(_core.Button, {
            size: "small",
            color: "secondary",
            children: "Share"
          }), (0, _jsxRuntime.jsx)(_core.Button, {
            size: "small",
            color: "secondary",
            children: "Learn More"
          })]
        })]
      })
    })]
  });
};

exports.Default = Default;
;

(function () {
  var reactHotLoader = typeof reactHotLoaderGlobal !== 'undefined' ? reactHotLoaderGlobal.default : undefined;

  if (!reactHotLoader) {
    return;
  }

  reactHotLoader.register(Default, "Default", "/var/jenkins_home/workspace/ish_Fuel_React_Components_master/src/fuel-lab/Card/Card.stories.js");
  reactHotLoader.register(_default, "default", "/var/jenkins_home/workspace/ish_Fuel_React_Components_master/src/fuel-lab/Card/Card.stories.js");
})();

;

(function () {
  var leaveModule = typeof reactHotLoaderGlobal !== 'undefined' ? reactHotLoaderGlobal.leaveModule : undefined;
  leaveModule && leaveModule(module);
})();
