// @deere/fuel-react/src/Input/InputLabel.tsx
import React from "react";
import { withStyles } from '@material-ui/core/styles';
import MaterialInputLabel from '@material-ui/core/InputLabel';
import { InputLabelProps as MaterialInputLabelProps } from '@material-ui/core/InputLabel';

export interface InputLabelProps extends MaterialInputLabelProps { };

const StyledInputLabel = withStyles((theme) => ({
  root: {
    color: 'var(--fuel-color-gray-900)',
    fontSize: theme.typography.button.fontSize,
    fontWeight: theme.typography.button.fontWeight,
    '&.Mui-focused': {
      color: 'var(--fuel-color-text-primary)',
    },
    '&.MuiInputLabel-shrink': {
      transform: 'none',
      position: 'relative',
      height: 'auto',
      overflow: 'hidden',
      lineHeight: theme.spacing(5),
      whiteSpace: 'nowrap',
      textOverflow: 'ellipsis'
    },
    '&.Mui-error': {
      color: 'var(--fuel-color-text-primary)'
    },
    marginBottom: theme.spacing(1)
  },
  asterisk: {
    color: theme.palette.error.main,
  },
}))(MaterialInputLabel);

export const InputLabel = React.forwardRef<any, InputLabelProps>(
  (
    {
      children,
      ...props
    },
    ref
  ) => {
    return (
      <StyledInputLabel shrink ref={ref} {...props}>
        {children}
      </StyledInputLabel>
    );
  });
