// @deere/fuel-react/src/Input/Input.tsx
import React from "react";
import { withStyles } from '@material-ui/core/styles';
import MaterialInputBase from '@material-ui/core/InputBase';
import { InputBaseProps as MaterialInputProps } from '@material-ui/core/InputBase';

export interface InputProps extends MaterialInputProps { };

const StyledInput = withStyles((theme) => ({
  input: {
    borderRadius: theme.shape.borderRadius,
    position: 'relative',
    backgroundColor: 'var(--fuel-color-white)',
    border: `1px solid var(--fuel-color-border-dark)`,
    borderTopColor: '#a6a6a6',
    padding: 'var(--fuel-spacing-1) var(--fuel-spacing-3)',
    fontSize: theme.typography.body1.fontSize,
    height: '1.625rem',
    '&:focus': {
      boxShadow: "var(--fuel-color-focused) 0px 0px 0px 1px",
      borderColor: 'var(--fuel-color-focused)'
    },
    "-webkit-appearance": "none",
    "&::placeholder": {
      color: "var(--fuel-color-text-tertiary)",
      opacity: "1"
    },
    "&[disabled]": {
      cursor: "not-allowed",
      backgroundColor: 'var(--fuel-color-gray-100)',
      borderColor: 'var(--fuel-color-border-light)'
    },
    "&[readonly]": {
      backgroundColor: 'var(--fuel-color-gray-100)'
    }
  },
  error: {
    '& .MuiInputBase-input': {
      border: `1px solid ${theme.palette.error.main}`,
      '&:focus': {
        boxShadow: `${theme.palette.error.main} 0px 0px 0px 1px`,
        borderColor: theme.palette.error.main
      },
    }
  }
}))(MaterialInputBase);

export const Input = React.forwardRef<any, InputProps>(
  (
    {
      children,
      ...props
    },
    ref
  ) => {
    return (
      <StyledInput ref={ref} {...props}>
        {children}
      </StyledInput>
    );
  });
