// @deere/fuel-react/lib/Radio/Radio.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.Radio = void 0;

var _react = _interopRequireDefault(require("react"));

var _Radio = _interopRequireDefault(require("@material-ui/core/Radio"));

var _styles = require("@material-ui/core/styles");

var _jsxRuntime = require("@emotion/react/jsx-runtime");

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

(function () {
  var enterModule = typeof reactHotLoaderGlobal !== 'undefined' ? reactHotLoaderGlobal.enterModule : undefined;
  enterModule && enterModule(module);
})();

function ownKeys(object, enumerableOnly) { var keys = Object.keys(object); if (Object.getOwnPropertySymbols) { var symbols = Object.getOwnPropertySymbols(object); if (enumerableOnly) { symbols = symbols.filter(function (sym) { return Object.getOwnPropertyDescriptor(object, sym).enumerable; }); } keys.push.apply(keys, symbols); } return keys; }

function _objectSpread(target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i] != null ? arguments[i] : {}; if (i % 2) { ownKeys(Object(source), true).forEach(function (key) { _defineProperty(target, key, source[key]); }); } else if (Object.getOwnPropertyDescriptors) { Object.defineProperties(target, Object.getOwnPropertyDescriptors(source)); } else { ownKeys(Object(source)).forEach(function (key) { Object.defineProperty(target, key, Object.getOwnPropertyDescriptor(source, key)); }); } } return target; }

function _defineProperty(obj, key, value) { if (key in obj) { Object.defineProperty(obj, key, { value: value, enumerable: true, configurable: true, writable: true }); } else { obj[key] = value; } return obj; }

function _extends() { _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; }; return _extends.apply(this, arguments); }

var __signature__ = typeof reactHotLoaderGlobal !== 'undefined' ? reactHotLoaderGlobal["default"].signature : function (a) {
  return a;
};

var StyledRadio = (0, _styles.withStyles)(function (theme) {
  return {
    root: {
      height: "1.875rem",
      width: "2.125rem",
      color: 'var(--fuel-color-border-dark)',
      marginLeft: theme.spacing(1),
      "&.Mui-focusVisible .MuiSvgIcon-root": {
        borderRadius: '50%',
        boxShadow: '0 0 0 0.125rem var(--fuel-color-focused)'
      },
      '&:hover': {
        backgroundColor: 'transparent'
      },
      '&.Mui-checked:hover': {
        backgroundColor: 'transparent'
      }
    }
  };
})(_Radio["default"]);

var Radio = /*#__PURE__*/_react["default"].forwardRef(function (_ref, ref) {
  var props = _extends({}, _ref);

  return (0, _jsxRuntime.jsx)(StyledRadio, _objectSpread({
    ref: ref
  }, props));
});

exports.Radio = Radio;
;

(function () {
  var reactHotLoader = typeof reactHotLoaderGlobal !== 'undefined' ? reactHotLoaderGlobal.default : undefined;

  if (!reactHotLoader) {
    return;
  }

  reactHotLoader.register(StyledRadio, "StyledRadio", "/var/jenkins_home/workspace/ish_Fuel_React_Components_master/src/Radio/Radio.tsx");
  reactHotLoader.register(Radio, "Radio", "/var/jenkins_home/workspace/ish_Fuel_React_Components_master/src/Radio/Radio.tsx");
})();

;

(function () {
  var leaveModule = typeof reactHotLoaderGlobal !== 'undefined' ? reactHotLoaderGlobal.leaveModule : undefined;
  leaveModule && leaveModule(module);
})();
