// @deere/ux.uxframe-react/lib/components/Collapse/Collapse.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.Collapse = void 0;

var _react = _interopRequireDefault(require("react"));

var _propTypes = _interopRequireDefault(require("prop-types"));

var _Collapse = _interopRequireDefault(require("react-bootstrap/Collapse"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _objectWithoutProperties(source, excluded) { if (source == null) return {}; var target = _objectWithoutPropertiesLoose(source, excluded); var key, i; if (Object.getOwnPropertySymbols) { var sourceSymbolKeys = Object.getOwnPropertySymbols(source); for (i = 0; i < sourceSymbolKeys.length; i++) { key = sourceSymbolKeys[i]; if (excluded.indexOf(key) >= 0) continue; if (!Object.prototype.propertyIsEnumerable.call(source, key)) continue; target[key] = source[key]; } } return target; }

function _objectWithoutPropertiesLoose(source, excluded) { if (source == null) return {}; var target = {}; var sourceKeys = Object.keys(source); var key, i; for (i = 0; i < sourceKeys.length; i++) { key = sourceKeys[i]; if (excluded.indexOf(key) >= 0) continue; target[key] = source[key]; } return target; }

var Collapse = function Collapse(_ref) {
  var children = _ref.children,
      props = _objectWithoutProperties(_ref, ["children"]);

  return /*#__PURE__*/_react["default"].createElement(_Collapse["default"], props, children);
};

exports.Collapse = Collapse;
Collapse.propTypes = {
  /**
   * Show the component; triggers the expand or collapse animation
   */
  "in": _propTypes["default"].bool,

  /**
   * Wait until the first "enter" transition to mount the component (add it to the DOM)
   */
  mountOnEnter: _propTypes["default"].bool,

  /**
   * Unmount the component (remove it from the DOM) when it is collapsed
   */
  unmountOnExit: _propTypes["default"].bool,

  /**
   * Run the expand animation when the component mounts, if it is initially
   * shown
   */
  appear: _propTypes["default"].bool,

  /**
   * Duration of the collapse animation in milliseconds, to ensure that
   * finishing callbacks are fired even if the original browser transition end
   * events are canceled
   */
  timeout: _propTypes["default"].number,

  /**
   * Callback fired before the component expands
   */
  onEnter: _propTypes["default"].func,

  /**
   * Callback fired after the component starts to expand
   */
  onEntering: _propTypes["default"].func,

  /**
   * Callback fired after the component has expanded
   */
  onEntered: _propTypes["default"].func,

  /**
   * Callback fired before the component collapses
   */
  onExit: _propTypes["default"].func,

  /**
   * Callback fired after the component starts to collapse
   */
  onExiting: _propTypes["default"].func,

  /**
   * Callback fired after the component has collapsed
   */
  onExited: _propTypes["default"].func,

  /**
   * The dimension used when collapsing, or a function that returns the
   * dimension
   *
   * _Note: Bootstrap only partially supports 'width'!
   * You will need to supply your own CSS animation for the `.width` CSS class._
   */
  dimension: _propTypes["default"].oneOfType([_propTypes["default"].oneOf(['height', 'width']), _propTypes["default"].func]),

  /**
   * Function that returns the height or width of the animating DOM node
   *
   * Allows for providing some custom logic for how much the Collapse component
   * should animate in its specified dimension. Called with the current
   * dimension prop value and the DOM node.
   *
   * @default element.offsetWidth | element.offsetHeight
   */
  getDimensionValue: _propTypes["default"].func,

  /**
   * ARIA role of collapsible element
   */
  role: _propTypes["default"].string
};
