// @deere/fuel-react/lib/Select/Select.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.Select = void 0;

var _react = _interopRequireDefault(require("react"));

var _Select = _interopRequireDefault(require("@material-ui/core/Select"));

var _styles = require("@material-ui/core/styles");

var _ = require("../");

var _KeyboardArrowDown = _interopRequireDefault(require("@material-ui/icons/KeyboardArrowDown"));

var _jsxRuntime = require("@emotion/react/jsx-runtime");

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

(function () {
  var enterModule = typeof reactHotLoaderGlobal !== 'undefined' ? reactHotLoaderGlobal.enterModule : undefined;
  enterModule && enterModule(module);
})();

function ownKeys(object, enumerableOnly) { var keys = Object.keys(object); if (Object.getOwnPropertySymbols) { var symbols = Object.getOwnPropertySymbols(object); if (enumerableOnly) { symbols = symbols.filter(function (sym) { return Object.getOwnPropertyDescriptor(object, sym).enumerable; }); } keys.push.apply(keys, symbols); } return keys; }

function _objectSpread(target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i] != null ? arguments[i] : {}; if (i % 2) { ownKeys(Object(source), true).forEach(function (key) { _defineProperty(target, key, source[key]); }); } else if (Object.getOwnPropertyDescriptors) { Object.defineProperties(target, Object.getOwnPropertyDescriptors(source)); } else { ownKeys(Object(source)).forEach(function (key) { Object.defineProperty(target, key, Object.getOwnPropertyDescriptor(source, key)); }); } } return target; }

function _defineProperty(obj, key, value) { if (key in obj) { Object.defineProperty(obj, key, { value: value, enumerable: true, configurable: true, writable: true }); } else { obj[key] = value; } return obj; }

function _objectWithoutProperties(source, excluded) { if (source == null) return {}; var target = _objectWithoutPropertiesLoose(source, excluded); var key, i; if (Object.getOwnPropertySymbols) { var sourceSymbolKeys = Object.getOwnPropertySymbols(source); for (i = 0; i < sourceSymbolKeys.length; i++) { key = sourceSymbolKeys[i]; if (excluded.indexOf(key) >= 0) continue; if (!Object.prototype.propertyIsEnumerable.call(source, key)) continue; target[key] = source[key]; } } return target; }

function _objectWithoutPropertiesLoose(source, excluded) { if (source == null) return {}; var target = {}; var sourceKeys = Object.keys(source); var key, i; for (i = 0; i < sourceKeys.length; i++) { key = sourceKeys[i]; if (excluded.indexOf(key) >= 0) continue; target[key] = source[key]; } return target; }

var __signature__ = typeof reactHotLoaderGlobal !== 'undefined' ? reactHotLoaderGlobal["default"].signature : function (a) {
  return a;
};

;
var StyledSelect = (0, _styles.withStyles)(function (theme) {
  return {
    select: {
      borderRadius: theme.shape.borderRadius,
      '&:focus': {
        borderRadius: theme.shape.borderRadius
      },
      paddingRight: '1.75rem !important',
      height: '1.625rem',
      display: 'inline-flex',
      alignItems: 'center',
      lineHeight: '1.5rem',
      "&[disabled]": {
        cursor: "not-allowed",
        backgroundColor: 'var(--fuel-color-gray-100)',
        borderColor: 'var(--fuel-color-border-light)'
      }
    },
    icon: {
      width: '.875em',
      height: '.875em',
      right: '.375rem',
      top: 'calc(50% - 11px)',
      color: 'currentColor'
    },
    iconOpen: {
      transform: 'none'
    }
  };
})(_Select["default"]);
var defaultMenuProps = {
  anchorOrigin: {
    vertical: "bottom",
    horizontal: "left"
  },
  transformOrigin: {
    vertical: 'top',
    horizontal: 'left'
  },
  getContentAnchorEl: null,
  disableScrollLock: true
};

var Select = /*#__PURE__*/_react["default"].forwardRef(function (_ref, ref) {
  var children = _ref.children,
      _ref$MenuProps = _ref.MenuProps,
      MenuProps = _ref$MenuProps === void 0 ? defaultMenuProps : _ref$MenuProps,
      props = _objectWithoutProperties(_ref, ["children", "MenuProps"]);

  return (0, _jsxRuntime.jsx)(StyledSelect, _objectSpread(_objectSpread({
    input: (0, _jsxRuntime.jsx)(_.Input, {}),
    IconComponent: _KeyboardArrowDown["default"],
    ref: ref,
    MenuProps: MenuProps
  }, props), {}, {
    children: children
  }));
});

exports.Select = Select;
;

(function () {
  var reactHotLoader = typeof reactHotLoaderGlobal !== 'undefined' ? reactHotLoaderGlobal.default : undefined;

  if (!reactHotLoader) {
    return;
  }

  reactHotLoader.register(StyledSelect, "StyledSelect", "/var/jenkins_home/workspace/ish_Fuel_React_Components_master/src/Select/Select.tsx");
  reactHotLoader.register(defaultMenuProps, "defaultMenuProps", "/var/jenkins_home/workspace/ish_Fuel_React_Components_master/src/Select/Select.tsx");
  reactHotLoader.register(Select, "Select", "/var/jenkins_home/workspace/ish_Fuel_React_Components_master/src/Select/Select.tsx");
})();

;

(function () {
  var leaveModule = typeof reactHotLoaderGlobal !== 'undefined' ? reactHotLoaderGlobal.leaveModule : undefined;
  leaveModule && leaveModule(module);
})();
