// @deere/fuel-react/lib/fuel-lab/Dialog/Dialog.stories.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.Default = exports["default"] = void 0;

var _react = _interopRequireDefault(require("react"));

var _ = require(".");

var _Button = require("../../Button/Button");

var _styles = require("@material-ui/core/styles");

var _IconButton = require("../../Button/IconButton");

var _Close = _interopRequireDefault(require("@material-ui/icons/Close"));

var _Typography = require("../../Typography/Typography");

var _jsxRuntime = require("@emotion/react/jsx-runtime");

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

(function () {
  var enterModule = typeof reactHotLoaderGlobal !== 'undefined' ? reactHotLoaderGlobal.enterModule : undefined;
  enterModule && enterModule(module);
})();

function _slicedToArray(arr, i) { return _arrayWithHoles(arr) || _iterableToArrayLimit(arr, i) || _unsupportedIterableToArray(arr, i) || _nonIterableRest(); }

function _nonIterableRest() { throw new TypeError("Invalid attempt to destructure non-iterable instance.\nIn order to be iterable, non-array objects must have a [Symbol.iterator]() method."); }

function _unsupportedIterableToArray(o, minLen) { if (!o) return; if (typeof o === "string") return _arrayLikeToArray(o, minLen); var n = Object.prototype.toString.call(o).slice(8, -1); if (n === "Object" && o.constructor) n = o.constructor.name; if (n === "Map" || n === "Set") return Array.from(o); if (n === "Arguments" || /^(?:Ui|I)nt(?:8|16|32)(?:Clamped)?Array$/.test(n)) return _arrayLikeToArray(o, minLen); }

function _arrayLikeToArray(arr, len) { if (len == null || len > arr.length) len = arr.length; for (var i = 0, arr2 = new Array(len); i < len; i++) { arr2[i] = arr[i]; } return arr2; }

function _iterableToArrayLimit(arr, i) { var _i = arr && (typeof Symbol !== "undefined" && arr[Symbol.iterator] || arr["@@iterator"]); if (_i == null) return; var _arr = []; var _n = true; var _d = false; var _s, _e; try { for (_i = _i.call(arr); !(_n = (_s = _i.next()).done); _n = true) { _arr.push(_s.value); if (i && _arr.length === i) break; } } catch (err) { _d = true; _e = err; } finally { try { if (!_n && _i["return"] != null) _i["return"](); } finally { if (_d) throw _e; } } return _arr; }

function _arrayWithHoles(arr) { if (Array.isArray(arr)) return arr; }

function ownKeys(object, enumerableOnly) { var keys = Object.keys(object); if (Object.getOwnPropertySymbols) { var symbols = Object.getOwnPropertySymbols(object); if (enumerableOnly) { symbols = symbols.filter(function (sym) { return Object.getOwnPropertyDescriptor(object, sym).enumerable; }); } keys.push.apply(keys, symbols); } return keys; }

function _objectSpread(target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i] != null ? arguments[i] : {}; if (i % 2) { ownKeys(Object(source), true).forEach(function (key) { _defineProperty(target, key, source[key]); }); } else if (Object.getOwnPropertyDescriptors) { Object.defineProperties(target, Object.getOwnPropertyDescriptors(source)); } else { ownKeys(Object(source)).forEach(function (key) { Object.defineProperty(target, key, Object.getOwnPropertyDescriptor(source, key)); }); } } return target; }

function _defineProperty(obj, key, value) { if (key in obj) { Object.defineProperty(obj, key, { value: value, enumerable: true, configurable: true, writable: true }); } else { obj[key] = value; } return obj; }

function _objectWithoutProperties(source, excluded) { if (source == null) return {}; var target = _objectWithoutPropertiesLoose(source, excluded); var key, i; if (Object.getOwnPropertySymbols) { var sourceSymbolKeys = Object.getOwnPropertySymbols(source); for (i = 0; i < sourceSymbolKeys.length; i++) { key = sourceSymbolKeys[i]; if (excluded.indexOf(key) >= 0) continue; if (!Object.prototype.propertyIsEnumerable.call(source, key)) continue; target[key] = source[key]; } } return target; }

function _objectWithoutPropertiesLoose(source, excluded) { if (source == null) return {}; var target = {}; var sourceKeys = Object.keys(source); var key, i; for (i = 0; i < sourceKeys.length; i++) { key = sourceKeys[i]; if (excluded.indexOf(key) >= 0) continue; target[key] = source[key]; } return target; }

var __signature__ = typeof reactHotLoaderGlobal !== 'undefined' ? reactHotLoaderGlobal["default"].signature : function (a) {
  return a;
};

var _default = {
  title: 'Components/Dialog',
  component: _.Dialog
};
var _default2 = _default;
exports["default"] = _default2;

var styles = function styles(theme) {
  return {
    root: {
      margin: 0,
      padding: theme.spacing(4)
    },
    closeButton: {
      position: 'absolute',
      right: theme.spacing(1),
      top: theme.spacing(1),
      color: theme.palette.grey[500]
    }
  };
};

var DialogTitle = (0, _styles.withStyles)(styles)(function (props) {
  var children = props.children,
      classes = props.classes,
      onClose = props.onClose,
      other = _objectWithoutProperties(props, ["children", "classes", "onClose"]);

  return (0, _jsxRuntime.jsxs)(_.DialogTitle, _objectSpread(_objectSpread({
    disableTypography: true,
    className: classes.root
  }, other), {}, {
    children: [(0, _jsxRuntime.jsx)(_Typography.Typography, {
      variant: "h6",
      children: children
    }), onClose ? (0, _jsxRuntime.jsx)(_IconButton.IconButton, {
      "aria-label": "close",
      className: classes.closeButton,
      onClick: onClose,
      children: (0, _jsxRuntime.jsx)(_Close["default"], {})
    }) : null]
  }));
});
var DialogContent = (0, _styles.withStyles)(function (theme) {
  return {
    root: {
      padding: theme.spacing(4)
    }
  };
})(_.DialogContent);
var DialogActions = (0, _styles.withStyles)(function (theme) {
  return {
    root: {
      margin: 0,
      padding: theme.spacing(4)
    }
  };
})(_.DialogActions);

var Default = function Default() {
  var _React$useState = _react["default"].useState(false),
      _React$useState2 = _slicedToArray(_React$useState, 2),
      open = _React$useState2[0],
      setOpen = _React$useState2[1];

  var handleClickOpen = function handleClickOpen() {
    setOpen(true);
  };

  var handleClose = function handleClose() {
    setOpen(false);
  };

  return (0, _jsxRuntime.jsxs)("div", {
    children: [(0, _jsxRuntime.jsx)(_Button.Button, {
      variant: "outlined",
      color: "secondary",
      onClick: handleClickOpen,
      children: "Open dialog"
    }), (0, _jsxRuntime.jsxs)(_.Dialog, {
      onClose: handleClose,
      "aria-labelledby": "customized-dialog-title",
      open: open,
      children: [(0, _jsxRuntime.jsx)(DialogTitle, {
        id: "customized-dialog-title",
        onClose: handleClose,
        children: "Modal title"
      }), (0, _jsxRuntime.jsxs)(DialogContent, {
        dividers: true,
        children: [(0, _jsxRuntime.jsx)(_Typography.Typography, {
          gutterBottom: true,
          children: "Cras mattis consectetur purus sit amet fermentum. Cras justo odio, dapibus ac facilisis in, egestas eget quam. Morbi leo risus, porta ac consectetur ac, vestibulum at eros."
        }), (0, _jsxRuntime.jsx)(_Typography.Typography, {
          gutterBottom: true,
          children: "Praesent commodo cursus magna, vel scelerisque nisl consectetur et. Vivamus sagittis lacus vel augue laoreet rutrum faucibus dolor auctor."
        }), (0, _jsxRuntime.jsx)(_Typography.Typography, {
          gutterBottom: true,
          children: "Aenean lacinia bibendum nulla sed consectetur. Praesent commodo cursus magna, vel scelerisque nisl consectetur et. Donec sed odio dui. Donec ullamcorper nulla non metus auctor fringilla."
        })]
      }), (0, _jsxRuntime.jsxs)(DialogActions, {
        children: [(0, _jsxRuntime.jsx)(_Button.Button, {
          onClick: handleClose,
          variant: "outlined",
          children: "Cancel"
        }), (0, _jsxRuntime.jsx)(_Button.Button, {
          autoFocus: true,
          onClick: handleClose,
          variant: "contained",
          color: "primary",
          children: "Save changes"
        })]
      })]
    })]
  });
};

exports.Default = Default;

__signature__(Default, "useState{[open, setOpen](false)}");

;

(function () {
  var reactHotLoader = typeof reactHotLoaderGlobal !== 'undefined' ? reactHotLoaderGlobal.default : undefined;

  if (!reactHotLoader) {
    return;
  }

  reactHotLoader.register(styles, "styles", "/var/jenkins_home/workspace/ish_Fuel_React_Components_master/src/fuel-lab/Dialog/Dialog.stories.js");
  reactHotLoader.register(DialogTitle, "DialogTitle", "/var/jenkins_home/workspace/ish_Fuel_React_Components_master/src/fuel-lab/Dialog/Dialog.stories.js");
  reactHotLoader.register(DialogContent, "DialogContent", "/var/jenkins_home/workspace/ish_Fuel_React_Components_master/src/fuel-lab/Dialog/Dialog.stories.js");
  reactHotLoader.register(DialogActions, "DialogActions", "/var/jenkins_home/workspace/ish_Fuel_React_Components_master/src/fuel-lab/Dialog/Dialog.stories.js");
  reactHotLoader.register(Default, "Default", "/var/jenkins_home/workspace/ish_Fuel_React_Components_master/src/fuel-lab/Dialog/Dialog.stories.js");
  reactHotLoader.register(_default, "default", "/var/jenkins_home/workspace/ish_Fuel_React_Components_master/src/fuel-lab/Dialog/Dialog.stories.js");
})();

;

(function () {
  var leaveModule = typeof reactHotLoaderGlobal !== 'undefined' ? reactHotLoaderGlobal.leaveModule : undefined;
  leaveModule && leaveModule(module);
})();
