// @deere/fuel-react/lib/fuel-lab/Header/Header.js
"use strict";

function _typeof(obj) { "@babel/helpers - typeof"; if (typeof Symbol === "function" && typeof Symbol.iterator === "symbol") { _typeof = function _typeof(obj) { return typeof obj; }; } else { _typeof = function _typeof(obj) { return obj && typeof Symbol === "function" && obj.constructor === Symbol && obj !== Symbol.prototype ? "symbol" : typeof obj; }; } return _typeof(obj); }

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.Header = void 0;

var _react = _interopRequireWildcard(require("react"));

var _core = require("@material-ui/core");

var _Menu = _interopRequireDefault(require("@material-ui/icons/Menu"));

var _jsxRuntime = require("@emotion/react/jsx-runtime");

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _getRequireWildcardCache(nodeInterop) { if (typeof WeakMap !== "function") return null; var cacheBabelInterop = new WeakMap(); var cacheNodeInterop = new WeakMap(); return (_getRequireWildcardCache = function _getRequireWildcardCache(nodeInterop) { return nodeInterop ? cacheNodeInterop : cacheBabelInterop; })(nodeInterop); }

function _interopRequireWildcard(obj, nodeInterop) { if (!nodeInterop && obj && obj.__esModule) { return obj; } if (obj === null || _typeof(obj) !== "object" && typeof obj !== "function") { return { "default": obj }; } var cache = _getRequireWildcardCache(nodeInterop); if (cache && cache.has(obj)) { return cache.get(obj); } var newObj = {}; var hasPropertyDescriptor = Object.defineProperty && Object.getOwnPropertyDescriptor; for (var key in obj) { if (key !== "default" && Object.prototype.hasOwnProperty.call(obj, key)) { var desc = hasPropertyDescriptor ? Object.getOwnPropertyDescriptor(obj, key) : null; if (desc && (desc.get || desc.set)) { Object.defineProperty(newObj, key, desc); } else { newObj[key] = obj[key]; } } } newObj["default"] = obj; if (cache) { cache.set(obj, newObj); } return newObj; }

(function () {
  var enterModule = typeof reactHotLoaderGlobal !== 'undefined' ? reactHotLoaderGlobal.enterModule : undefined;
  enterModule && enterModule(module);
})();

function _extends() { _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; }; return _extends.apply(this, arguments); }

function _slicedToArray(arr, i) { return _arrayWithHoles(arr) || _iterableToArrayLimit(arr, i) || _unsupportedIterableToArray(arr, i) || _nonIterableRest(); }

function _nonIterableRest() { throw new TypeError("Invalid attempt to destructure non-iterable instance.\nIn order to be iterable, non-array objects must have a [Symbol.iterator]() method."); }

function _unsupportedIterableToArray(o, minLen) { if (!o) return; if (typeof o === "string") return _arrayLikeToArray(o, minLen); var n = Object.prototype.toString.call(o).slice(8, -1); if (n === "Object" && o.constructor) n = o.constructor.name; if (n === "Map" || n === "Set") return Array.from(o); if (n === "Arguments" || /^(?:Ui|I)nt(?:8|16|32)(?:Clamped)?Array$/.test(n)) return _arrayLikeToArray(o, minLen); }

function _arrayLikeToArray(arr, len) { if (len == null || len > arr.length) len = arr.length; for (var i = 0, arr2 = new Array(len); i < len; i++) { arr2[i] = arr[i]; } return arr2; }

function _iterableToArrayLimit(arr, i) { var _i = arr && (typeof Symbol !== "undefined" && arr[Symbol.iterator] || arr["@@iterator"]); if (_i == null) return; var _arr = []; var _n = true; var _d = false; var _s, _e; try { for (_i = _i.call(arr); !(_n = (_s = _i.next()).done); _n = true) { _arr.push(_s.value); if (i && _arr.length === i) break; } } catch (err) { _d = true; _e = err; } finally { try { if (!_n && _i["return"] != null) _i["return"](); } finally { if (_d) throw _e; } } return _arr; }

function _arrayWithHoles(arr) { if (Array.isArray(arr)) return arr; }

function ownKeys(object, enumerableOnly) { var keys = Object.keys(object); if (Object.getOwnPropertySymbols) { var symbols = Object.getOwnPropertySymbols(object); if (enumerableOnly) { symbols = symbols.filter(function (sym) { return Object.getOwnPropertyDescriptor(object, sym).enumerable; }); } keys.push.apply(keys, symbols); } return keys; }

function _objectSpread(target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i] != null ? arguments[i] : {}; if (i % 2) { ownKeys(Object(source), true).forEach(function (key) { _defineProperty(target, key, source[key]); }); } else if (Object.getOwnPropertyDescriptors) { Object.defineProperties(target, Object.getOwnPropertyDescriptors(source)); } else { ownKeys(Object(source)).forEach(function (key) { Object.defineProperty(target, key, Object.getOwnPropertyDescriptor(source, key)); }); } } return target; }

function _defineProperty(obj, key, value) { if (key in obj) { Object.defineProperty(obj, key, { value: value, enumerable: true, configurable: true, writable: true }); } else { obj[key] = value; } return obj; }

function _objectWithoutProperties(source, excluded) { if (source == null) return {}; var target = _objectWithoutPropertiesLoose(source, excluded); var key, i; if (Object.getOwnPropertySymbols) { var sourceSymbolKeys = Object.getOwnPropertySymbols(source); for (i = 0; i < sourceSymbolKeys.length; i++) { key = sourceSymbolKeys[i]; if (excluded.indexOf(key) >= 0) continue; if (!Object.prototype.propertyIsEnumerable.call(source, key)) continue; target[key] = source[key]; } } return target; }

function _objectWithoutPropertiesLoose(source, excluded) { if (source == null) return {}; var target = {}; var sourceKeys = Object.keys(source); var key, i; for (i = 0; i < sourceKeys.length; i++) { key = sourceKeys[i]; if (excluded.indexOf(key) >= 0) continue; target[key] = source[key]; } return target; }

var __signature__ = typeof reactHotLoaderGlobal !== 'undefined' ? reactHotLoaderGlobal["default"].signature : function (a) {
  return a;
};

var useStyles = (0, _core.makeStyles)(function () {
  return {
    header: {
      backgroundColor: "#fff",
      boxShadow: 'none',
      borderBottom: '1px solid #ddd'
    },
    logoContainer: {
      minHeight: "56px",
      display: "flex",
      alignItems: "center",
      "&:before": {
        background: "#367c2b",
        content: '""',
        height: "100%",
        left: 0,
        position: "absolute",
        top: 0,
        width: ".75rem"
      },
      "&:after": {
        background: "#ffde00",
        content: '""',
        height: "100%",
        left: ".75rem",
        position: "absolute",
        top: 0,
        width: ".25rem"
      }
    },
    logo: {
      width: "160px",
      height: "30px",
      marginLeft: "1.5rem"
    },
    toolbar: {
      minHeight: 56,
      borderBottom: '1px solid #ddd'
    },
    menuButton: {
      marginRight: "8px",
      marginLeft: "auto"
    },
    navItems: {
      display: "flex"
    }
  };
});

var Logo = function Logo(_ref) {
  var linkProps = _ref.linkProps,
      props = _objectWithoutProperties(_ref, ["linkProps"]);

  var classes = useStyles();
  return (0, _jsxRuntime.jsx)(_core.Link, _objectSpread(_objectSpread({}, linkProps), {}, {
    children: (0, _jsxRuntime.jsx)(_core.Box, {
      className: classes.logoContainer,
      children: (0, _jsxRuntime.jsx)("svg", {
        className: classes.logo,
        xmlns: "http://www.w3.org/2000/svg",
        width: "192",
        height: "36",
        viewBox: "0 0 192 36",
        children: (0, _jsxRuntime.jsxs)("g", {
          fill: "none",
          children: [(0, _jsxRuntime.jsx)("path", {
            fill: "#367C2B",
            d: "M110.564145,26.255 L110.564145,13.033 L107.529948,13.033 L107.529948,21.956 L99.6251192,13.033 L96.619772,13.033 L96.619772,26.255 L99.6519793,26.255 L99.6519793,17.243 L107.530943,26.255 L110.564145,26.255 Z M173.177036,18.721 C173.799793,18.721 174.852311,18.521 174.852311,17.213 C174.852311,16.276 174.240497,15.74 173.177036,15.74 L167.637886,15.74 L167.637886,18.713 L173.177036,18.721 Z M178.352083,26.255 L174.686176,26.255 L171.030218,21.419 L167.471751,21.419 L167.471751,26.255 L164.440539,26.255 L164.440539,13.033 L173.010902,13.033 C175.960539,13.033 177.714404,14.623 177.714404,17.279 C177.714404,18.869 176.838964,20.659 174.540933,21.216 L178.352083,26.255 Z M125.43171,9.743 C131.237472,9.743 134.710383,12.88 134.710383,18.12 C134.710383,21.877 132.280041,26.255 125.43171,26.255 L118.10885,26.255 L118.10885,9.743 L125.43171,9.743 Z M121.15001,12.473 L121.15001,23.531 L125.43171,23.531 C127.755606,23.531 131.661264,22.807 131.661264,17.951 C131.661264,14.469 129.391088,12.473 125.43171,12.473 L121.15001,12.473 Z M182.767088,23.552 L182.767088,21 L189.766632,21 L189.766632,18.295 L182.767088,18.295 L182.767088,15.74 L191.80999,15.74 L191.80999,13.033 L179.743834,13.033 L179.743834,26.255 L191.810984,26.255 L191.810984,23.552 L182.767088,23.552 Z M153.384124,23.552 L153.384124,21 L160.377699,21 L160.377699,18.295 L153.384124,18.295 L153.384124,15.74 L162.422052,15.74 L162.422052,13.033 L150.349927,13.033 L150.349927,26.255 L162.422052,26.255 L162.422052,23.552 L153.384124,23.552 Z M139.293513,23.552 L139.293513,21 L146.282114,21 L146.282114,18.295 L139.293513,18.295 L139.293513,15.74 L148.33144,15.74 L148.33144,13.033 L136.26429,13.033 L136.26429,26.255 L148.33144,26.255 L148.33144,23.552 L139.293513,23.552 Z M91.1243938,13.033 L91.1243938,18.295 L83.2364767,18.295 L83.2364767,13.033 L80.2062591,13.033 L80.2062591,26.255 L83.2364767,26.255 L83.2364767,21 L91.123399,21 L91.123399,26.255 L94.1516269,26.255 L94.1516269,13.033 L91.123399,13.033 L91.1243938,13.033 Z M70.6321244,23.867 C68.8613472,23.867 65.877886,23.312 65.877886,19.607 C65.877886,16.951 67.6148394,15.424 70.6321244,15.424 C73.651399,15.424 75.379399,16.951 75.379399,19.607 C75.379399,23.312 72.4048912,23.867 70.6321244,23.867 Z M62.8556269,19.747 C62.8556269,22.891 64.8910259,26.568 70.6321244,26.568 C76.3742176,26.568 78.412601,22.891 78.412601,19.747 C78.412601,15.349 75.5027565,12.719 70.6321244,12.719 C65.762487,12.719 62.8556269,15.349 62.8556269,19.747 Z M49.5310259,23.115 C49.5310259,23.115 51.2520622,23.74 53.1760415,23.74 C55.2313368,23.74 56.5106736,23.456 57.2259482,22.724 C57.8188601,22.123 58.0546321,21.117 58.0267772,19.634 L58.0267772,9.76 L61.0609741,9.76 L61.0609741,20.057 C61.0172021,24.62 58.8246218,26.568 53.7311503,26.568 C51.2540518,26.568 49.5310259,26.169 49.5310259,26.169 L49.5310259,23.115 Z M36.8978238,3.89 C31.8540933,1.405 26.0443523,0 19.9998342,0 C13.9304456,0 8.12965803,1.412 3.07398964,3.913 C1.23421718,8.38011068 0.289334062,13.1682122 0.293471503,18.003 C0.293471503,22.979 1.28033161,27.732 3.07100518,32.085 C8.12965803,34.591 13.9304456,36 19.9988394,36 C26.0443523,36 31.8550881,34.596 36.9008083,32.11 C38.7448489,27.6379995 39.6921328,22.8439475 39.6882902,18.003 C39.6882902,13.016 38.6974508,8.255 36.8988187,3.891 L36.8978238,3.89 Z"
          }), (0, _jsxRuntime.jsx)("path", {
            fill: "#FFDE00",
            d: "M14.6208497,13.919 C13.8080829,15.547 13.2420311,17.369 13.0112332,19.451 C12.9435855,19.459 12.5307358,19.495 12.5307358,19.495 C10.8315855,19.651 9.07573057,19.784 7.37260104,20.445 L7.49297409,21.137 C10.2705078,24.609 14.4775959,26.703 14.5193782,26.721 L14.6168705,26.617 L13.7076062,25.161 C13.4907358,24.997 12.9236891,24.867 12.9047876,24.864 C11.0106528,24.229 9.68157513,21.74 9.46172021,21.312 C11.0235855,21.307 12.3944456,21.762 13.7205389,22.202 C14.5959793,22.495 15.4196891,22.768 16.2752332,22.909 C17.7774093,23.151 19.3492228,22.867 20.7618653,22.691 C21.909886,22.552 23.0240829,22.555 24.1333057,22.628 C24.2656166,22.792 25.4325389,24.211 26.5447461,24.795 C27.9414715,25.531 32.0053057,26.749 32.2072539,26.82 C32.288829,26.899 33.8546736,28.481 34.8156684,30.315 C30.2623834,32.421 25.141057,33.536 19.9968497,33.536 C14.8397098,33.536 9.70047668,32.413 5.12829016,30.284 C3.62082975,26.3659959 2.84972485,22.2008221 2.85413472,18 C2.85413472,13.779 3.6211399,9.645 5.13027979,5.719 C9.70047668,3.591 14.8397098,2.467 19.9968497,2.467 C25.1500104,2.467 30.2862591,3.585 34.8435233,5.697 C36.3562761,9.62039341 37.1304579,13.7922308 37.1266321,18 C37.1266321,21.875 36.4859689,25.661 35.2245389,29.268 C33.7721036,26.738 33.1841658,25.972 33.1841658,25.972 C33.1652642,25.959 31.1338446,24.893 30.6304663,24.459 C30.2305492,24.109 30.0405389,23.399 29.8176995,22.576 C29.5112953,21.435 29.1253057,20.024 28.0240415,18.857 L29.2486632,18.355 C28.9074404,15.984 26.1169741,14.445 26.0861347,14.427 L25.9299482,14.503 C26.9963938,15.847 27.3067772,16.989 27.3067772,17.688 C27.3067772,18.041 27.2510674,18.278 27.2271917,18.375 C26.9058653,18.257 24.8436062,17.588 20.1560207,17.765 C19.4666114,17.789 18.4807461,17.757 17.8928083,17.183 C17.5605389,16.859 17.3934093,16.393 17.3934093,15.797 L17.3934093,15.792 C17.395399,14.156 18.0788394,11.953 18.1793161,11.642 L19.402943,11.792 C20.8573679,10.976 21.4293886,10.279 21.5079793,10.169 L21.4403316,10.096 C20.1958135,10.257 18.5682902,10.586 18.5682902,10.586 C18.6777202,10.3 19.0070052,9.359 19.0070052,8.295 C19.0070052,7.143 18.624,5.855 17.1705699,5.104 L17.0153782,5.18 C17.0273161,5.19 17.8609741,6.104 17.8609741,7.355 C17.8609741,7.7 17.7953161,8.076 17.6291813,8.464 C17.1546528,8.265 16.9596684,8.245 16.3906321,8.229 C16.1130777,7.251 15.2625078,6.416 14.1025492,5.995 L13.9632746,6.08 C13.9762073,6.088 15.2018238,6.967 15.2734508,8.263 C14.6536788,8.312 14.3910466,8.372 13.6290155,8.607 C13.3106736,8.071 12.5804767,7.685 11.6672332,7.576 L11.5488497,7.666 C11.5587979,7.672 12.6361865,8.046 12.8769326,8.877 C12.2094093,9.188 11.5170155,9.661 11.0464663,10.044 L11.092228,10.224 C13.8876684,9.134 16.4980725,8.814 17.1636062,9.609 C17.2760207,9.743 17.3327254,9.901 17.3327254,10.085 C17.3327254,10.32 17.2421969,10.588 17.0671088,10.899 C15.6833161,11.2 13.6150881,11.795 11.9159378,12.437 L12.1417617,13.153 C12.1417617,13.153 13.8518549,13.503 14.4994819,13.637 C14.4994819,13.637 14.7442073,13.672 14.6218446,13.919 L14.6208497,13.919 Z M35.6970777,4.956 C30.9856166,2.726 25.5936995,1.473 19.9958549,1.473 C14.3830881,1.473 9.00310881,2.735 4.27672539,4.976 C2.64245769,9.12526096 1.80562804,13.54837 1.81056995,18.011 C1.81056995,22.588 2.68103627,26.976 4.26876684,31.021 C8.9921658,33.268 14.3771192,34.528 19.9968497,34.528 C25.5946943,34.528 30.9925803,33.276 35.7050363,31.041 C37.3362998,26.8927045 38.1720543,22.4716224 38.1682073,18.011 C38.1682073,13.412 37.2927668,9.013 35.6960829,4.956 L35.6970777,4.956 Z"
          })]
        })
      })
    })
  }));
};

__signature__(Logo, "useStyles{classes}", function () {
  return [useStyles];
});

var SideDrawer = function SideDrawer(_ref2) {
  var navLinks = _ref2.navLinks;
  var classes = useStyles();

  var _useState = (0, _react.useState)({
    right: false
  }),
      _useState2 = _slicedToArray(_useState, 2),
      state = _useState2[0],
      setState = _useState2[1];

  var toggleDrawer = function toggleDrawer(anchor, open) {
    return function (event) {
      if (event.type === "keydown" && (event.key === "Tab" || event.key === "Shift")) {
        return;
      }

      setState(_defineProperty({}, anchor, open));
    };
  };

  var sideDrawerList = function sideDrawerList(anchor) {
    return (0, _jsxRuntime.jsx)("div", {
      className: classes.list,
      role: "presentation",
      onClick: toggleDrawer(anchor, false),
      onKeyDown: toggleDrawer(anchor, false),
      children: (0, _jsxRuntime.jsx)(_core.List, {
        component: "nav",
        children: navLinks.map(function (_ref3) {
          var title = _ref3.title,
              path = _ref3.path;
          return (0, _jsxRuntime.jsx)("a", {
            href: path,
            className: classes.linkText,
            children: (0, _jsxRuntime.jsx)(_core.ListItem, {
              button: true,
              children: (0, _jsxRuntime.jsx)(_core.ListItemText, {
                primary: title
              })
            })
          }, title);
        })
      })
    });
  };

  return (0, _jsxRuntime.jsxs)(_react["default"].Fragment, {
    children: [(0, _jsxRuntime.jsx)(_core.IconButton, {
      edge: "start",
      "aria-label": "menu",
      onClick: toggleDrawer("right", true),
      children: (0, _jsxRuntime.jsx)(_Menu["default"], {
        fontSize: "large",
        color: "#666"
      })
    }), (0, _jsxRuntime.jsx)(_core.Drawer, {
      anchor: "right",
      open: state.right,
      onOpen: toggleDrawer("right", true),
      onClose: toggleDrawer("right", false),
      children: sideDrawerList("right")
    })]
  });
};

__signature__(SideDrawer, "useStyles{classes}\nuseState{[state, setState]({ right: false })}", function () {
  return [useStyles];
});

var Header = function Header(_ref4) {
  var props = _extends({}, _ref4);

  var classes = useStyles();
  var navLinks = [{
    title: "about us",
    path: "/about-us"
  }, {
    title: "product",
    path: "/product"
  }, {
    title: "blog",
    path: "/blog"
  }, {
    title: "contact",
    path: "/contact"
  }, {
    title: "faq",
    path: "/faq"
  }];
  return (0, _jsxRuntime.jsxs)(_core.AppBar, {
    className: classes.header,
    children: [(0, _jsxRuntime.jsx)(_core.Toolbar, {
      className: classes.toolbar,
      children: (0, _jsxRuntime.jsx)(Logo, {
        linkProps: {
          href: "/hi"
        }
      })
    }), (0, _jsxRuntime.jsxs)(_core.Box, {
      children: [(0, _jsxRuntime.jsx)(_core.List, {
        component: "nav",
        className: classes.navItems,
        "aria-labelledby": "main navigation",
        children: navLinks.map(function (_ref5) {
          var title = _ref5.title,
              path = _ref5.path;
          return (0, _jsxRuntime.jsx)("a", {
            href: path,
            children: (0, _jsxRuntime.jsx)(_core.ListItem, {
              button: true,
              children: (0, _jsxRuntime.jsx)(_core.ListItemText, {
                primary: title
              })
            })
          }, title);
        })
      }), (0, _jsxRuntime.jsx)(SideDrawer, {
        navLinks: navLinks
      })]
    })]
  });
};

exports.Header = Header;

__signature__(Header, "useStyles{classes}", function () {
  return [useStyles];
});

;

(function () {
  var reactHotLoader = typeof reactHotLoaderGlobal !== 'undefined' ? reactHotLoaderGlobal.default : undefined;

  if (!reactHotLoader) {
    return;
  }

  reactHotLoader.register(useStyles, "useStyles", "/var/jenkins_home/workspace/ish_Fuel_React_Components_master/src/fuel-lab/Header/Header.js");
  reactHotLoader.register(Logo, "Logo", "/var/jenkins_home/workspace/ish_Fuel_React_Components_master/src/fuel-lab/Header/Header.js");
  reactHotLoader.register(SideDrawer, "SideDrawer", "/var/jenkins_home/workspace/ish_Fuel_React_Components_master/src/fuel-lab/Header/Header.js");
  reactHotLoader.register(Header, "Header", "/var/jenkins_home/workspace/ish_Fuel_React_Components_master/src/fuel-lab/Header/Header.js");
})();

;

(function () {
  var leaveModule = typeof reactHotLoaderGlobal !== 'undefined' ? reactHotLoaderGlobal.leaveModule : undefined;
  leaveModule && leaveModule(module);
})();
