// @deere/fuel-react/src/Radio/Radio.tsx
import React from "react";
import MaterialRadio from '@material-ui/core/Radio';
import { withStyles } from '@material-ui/core/styles';
import { RadioProps as MaterialRadioProps } from '@material-ui/core/Radio';

export interface RadioProps extends Omit<MaterialRadioProps,
  'disableElevation' |
  'disableFocusRipple' |
  'disableRipple' |
  'centerRipple' |
  'disableTouchRipple' |
  'focusRipple' |
  'TouchRippleProps' |
  'color'
> {
}


const StyledRadio = withStyles((theme) => ({
  root: {
    height: "1.875rem",
    width: "2.125rem",
    color: 'var(--fuel-color-border-dark)',
    marginLeft: theme.spacing(1),
    "&.Mui-focusVisible .MuiSvgIcon-root": {
      borderRadius: '50%',
      boxShadow: '0 0 0 0.125rem var(--fuel-color-focused)'
    },
    '&:hover': {
      backgroundColor: 'transparent'
    },
    '&.Mui-checked:hover': {
      backgroundColor: 'transparent'
    }
  }
}))(MaterialRadio);

export const Radio = React.forwardRef<any, RadioProps>(
  (
    {
      ...props
    },
    ref
  ) => {
    return (
      <StyledRadio ref={ref} {...props} />
    );
  });
