// @deere/fuel-react/src/themes/baseTheme.ts
import { createMuiTheme } from '@material-ui/core/styles';
import {
  FuelFontWeightSemibold, FuelFontFamilyBase, FuelFontSizeHeading1,
  FuelLineHeight2xl, FuelFontSizeHeading2, FuelShadowMd, FuelShadowLg,
  FuelShadowXl, FuelShadow2xl, FuelLineHeightXl, FuelFontSizeHeading3,
  FuelLineHeightSm, FuelFontSizeHeading4, FuelFontSizeHeading5, FuelFontSizeHeading6,
  FuelLineHeightMd, FuelFontSizeMd, FuelColorTextTertiary, FuelFontSizeXs,
  FuelLineHeightXs, FuelFontSizeSm, FuelFontWeightRegular
} from '@deere/fuel-tokens/dist/web/jd-green/tokens';

export const baseTheme = createMuiTheme({
  typography: {
    fontFamily: FuelFontFamilyBase,
    fontWeightMedium: FuelFontWeightSemibold,
    h1: {
      fontWeight: FuelFontWeightSemibold,
      fontSize: FuelFontSizeHeading1,
      lineHeight: FuelLineHeight2xl,
    },
    h2: {
      fontWeight: FuelFontWeightSemibold,
      fontSize: FuelFontSizeHeading2,
      lineHeight: FuelLineHeightXl,
    },
    h3: {
      fontWeight: FuelFontWeightSemibold,
      fontSize: FuelFontSizeHeading3,
      lineHeight: FuelLineHeightXl,
    },
    h4: {
      fontWeight: FuelFontWeightSemibold,
      fontSize: FuelFontSizeHeading4,
      lineHeight: FuelLineHeightMd,
    },
    h5: {
      fontWeight: FuelFontWeightSemibold,
      fontSize: FuelFontSizeHeading5,
      lineHeight: FuelLineHeightSm,
    },
    h6: {
      fontWeight: FuelFontWeightSemibold,
      fontSize: FuelFontSizeHeading6,
      lineHeight: FuelLineHeightSm,
    },
    body1: {
      fontSize: FuelFontSizeMd
    },
    subtitle1: {
      color: FuelColorTextTertiary,
      fontWeight: FuelFontWeightSemibold,
      fontSize: FuelFontSizeSm,
      lineHeight: FuelLineHeightSm
    },
    subtitle2: {
      fontWeight: FuelFontWeightSemibold,
      color: FuelColorTextTertiary,
      fontSize: FuelFontSizeXs,
      lineHeight: FuelLineHeightXs
    },
    button: {
      textTransform: 'none',
      fontSize: FuelFontSizeSm,
      lineHeight: FuelLineHeightMd,
      fontWeight: FuelFontWeightSemibold
    },
    caption: {
      fontSize: FuelFontSizeXs,
      lineHeight: FuelLineHeightXs,
      fontWeight: FuelFontWeightRegular
    },
    overline: {
      fontSize: FuelFontSizeXs,
      lineHeight: FuelLineHeightXs,
      fontWeight: FuelFontWeightRegular
    },
  },
  props: {
    MuiButtonBase: {
      disableRipple: true
    },
    MuiButtonGroup: {
      disableRipple: true
    },
    MuiLink: {
      underline: "always"
    }
  },
  shape: {
    borderRadius: 4,
  },
  spacing: factor => `${0.25 * factor}rem`,
  shadows: [
    "none",
    FuelShadowMd,
    FuelShadowMd,
    FuelShadowMd,
    FuelShadowMd,
    FuelShadowLg,
    FuelShadowLg,
    FuelShadowLg,
    FuelShadowXl,
    FuelShadowXl,
    FuelShadowXl,
    FuelShadowXl,
    FuelShadowXl,
    FuelShadowXl,
    FuelShadowXl,
    FuelShadowXl,
    FuelShadow2xl,
    FuelShadow2xl,
    FuelShadow2xl,
    FuelShadow2xl,
    FuelShadow2xl,
    FuelShadow2xl,
    FuelShadow2xl,
    FuelShadow2xl,
    FuelShadow2xl
  ],
});
