// @deere/fuel-react/src/themes/jd-green.ts
import { createMuiTheme } from '@material-ui/core/styles';
import { baseTheme } from './baseTheme';
import { baseOverrides } from './baseOverrides';
import {
  FuelColorBrandPrimary, FuelColorTextPrimary, FuelColorTextDisabled, FuelColorTextSecondary,
  FuelColorBackgroundGray, FuelColorBorderLight, FuelColorBackgroundActionDefaultHovered,
  FuelColorBackgroundActionDefaultSelected, FuelColorGray50, FuelColorGray100, FuelColorGray200,
  FuelColorGray300, FuelColorGray400, FuelColorGray500, FuelColorGray600, FuelColorGray700,
  FuelColorGray800, FuelColorGray900
} from '@deere/fuel-tokens/dist/web/jd-green/tokens';
import '!style-loader!css-loader!@deere/fuel-tokens/dist/web/jd-green/fuel-theme-jd-green.css';

export const jdGreen = createMuiTheme({
  ...baseTheme,
  palette: {
    text: {
      primary: FuelColorTextPrimary,
      secondary: FuelColorTextSecondary,
      disabled: FuelColorTextDisabled
    },
    type: 'light',
    primary: {
      main: FuelColorBrandPrimary,
    },
    secondary: {
      main: FuelColorBrandPrimary,
    },
    background: {
      default: FuelColorBackgroundGray
    },
    divider: FuelColorBorderLight,
    action: {
      hover: FuelColorBackgroundActionDefaultHovered,
      selected: FuelColorBackgroundActionDefaultSelected,
    },
    error: {
      main: '#c21020'
    },
    grey: {
      50: FuelColorGray50,
      100: FuelColorGray100,
      200: FuelColorGray200,
      300: FuelColorGray300,
      400: FuelColorGray400,
      500: FuelColorGray500,
      600: FuelColorGray600,
      700: FuelColorGray700,
      800: FuelColorGray800,
      900: FuelColorGray900,
    }
  },
  overrides: {
    ...baseOverrides,
  }
});
